docs/codeql/ql-language-reference/expressions.rst

```diff
@@ -488,6 +488,100 @@ value for each value generated by the ``<formula>``:
   value generated by the ``<formula>``. Here, the aggregation function is applied to each of the 
   resulting combinations.
 
+Example of monotonic aggregates
+-------------------------------
+
+Consider this query:
+
+.. code-block:: ql
+
+   string getPerson() { result = "Alice" or
+                        result = "Bob" or
+                        result = "Charles" or
+                        result = "Diane"
+                      }
+   string getFruit(string p) { p = "Alice"   and result = "Orange" or
+                               p = "Alice"   and result = "Apple" or
+                               p = "Bob"     and result = "Apple" or
+                               p = "Charles" and result = "Apple" or
+                               p = "Charles" and result = "Banana"
+                             }
+   int getPrice(string f) { f = "Apple"  and result = 100 or
+                            f = "Orange" and result = 100 or
+                            f = "Orange" and result =   1
+                          }
+
+   predicate nonmono(string p, int cost) {
+     p = getPerson() and cost = sum(string f | f = getFruit(p) | getPrice(f))
+   }
+
+   language[monotonicAggregates]
+   predicate mono(string p, int cost) {
+     p = getPerson() and cost = sum(string f | f = getFruit(p) | getPrice(f))
+   }
+
+   from string variant, string person, int cost
+   where variant = "default"  and nonmono(person, cost) or
+         variant = "monotonic" and mono(person, cost)
+   select variant, person, cost
+   order by variant, person
+
+The query produces these results:
+
++-----------+---------+------+
+|  variant  | person  | cost |
++-----------+---------+------+
+| default   | Alice   | 201  |
+| default   | Bob     | 100  |
+| default   | Charles | 100  |
+| default   | Diane   | 0    |
+| monotonic | Alice   | 101  |
+| monotonic | Alice   | 200  |
+| monotonic | Bob     | 100  |
+| monotonic | Diane   | 0    |
++-----------+---------+------+
+
+The two variants of the aggregate semantics differ in what happens
+when ``getPrice(f)`` has multiple results.
```

Comment:
This should probably also mention that they also differ when `getPrice(f)` has zero results.

Reply:
Fair point, changed.